fields.h

```diff
@@ -76,9 +76,15 @@ String getPalette() {
 }
 
 String setPalette(String value) {
-  currentPaletteIndex = value.toInt();
-  if (currentPaletteIndex < 0) currentPaletteIndex = 0;
-  else if (currentPaletteIndex >= paletteCount) currentPaletteIndex = paletteCount - 1;
+  // value.toInt() returns long, while currentPaletteIndex is declared as uint8_t
+  long tmp = value.toInt();
+  // clamp to [0.. paletteCount-1]
+  if (tmp < 1) {
+    tmp = 1;
+  } else if (tmp > (paletteCount+1)) {
+    tmp = paletteCount+1;
+  }
+  currentPaletteIndex = tmp;
```

Comment:
Thanks for this pull request!  Shouldn't this actually clamp to `[0.. paletteCount-1]` (as noted in the comment( instead of `[1.. paletteCount+1]`?